openmc/opencg_compatible.py

```diff
@@ -881,16 +881,27 @@ def get_opencg_lattice(openmc_lattice):
     universes = openmc_lattice.universes
     outer = openmc_lattice.outer
 
+    # Convert 2D dimension to 3D for OpenCG
+    if len(dimension) == 2:
+        new_dimension = np.ones(3, dtype=np.int)
+        new_dimension[:2] = dimension
+        dimension = new_dimension
+
+    # Convert 2D pitch to 3D for OpenCG
     if len(pitch) == 2:
-        new_pitch = np.ones(3, dtype=np.float64) * np.inf
+        new_pitch = np.ones(3, dtype=np.float64) * np.finfo(np.float64).max
         new_pitch[:2] = pitch
         pitch = new_pitch
 
+    # Convert 2D lower left to 3D for OpenCG
     if len(lower_left) == 2:
-        new_lower_left = np.ones(3, dtype=np.float64)
+        new_lower_left = np.ones(3, dtype=np.float64) * np.finfo(np.float64).min
         new_lower_left[:2] = lower_left
         lower_left = new_lower_left
 
+    # Convert 2D universes array to 3D for OpenCG
+    universes = np.atleast_3d(universes)
```

Comment:
Let's say a user inputs a 2D lattice with universes:

    universes = [[1,2,3],
                 [4,5,6]]

The universes attribute will then have a shape of (2,3) with dimensions (3,2) in (x,y) notation. Notice that the shape is by ordered by (y,x). The `np.atleast_3d(universes)` function would then reconstruct universes to have a shape of (2,3,1). What we really want is the shape to be (1,3,2). So I think this line should really check if `universes` is 2D; if so, another dimension should be added such that the shape is (1,y,x).